src/query/tsdb/remote/server.go

```diff
@@ -107,18 +118,27 @@ func (s *grpcServer) Fetch(
 		return err
 	}
 
-	response, err := encodeToCompressedFetchResult(result, pools)
+	results, err := encodeToCompressedSeries(result, pools)
 	if err != nil {
 		logger.Error("unable to compress query", zap.Error(err))
 		return err
 	}
 
-	err = stream.Send(response)
-	if err != nil {
-		logger.Error("unable to send fetch result", zap.Error(err))
+	size := min(defaultBatch, len(results))
+	for ; len(results) >= size; results = results[size:] {
+		response := &rpc.FetchResponse{
+			Series: results[:size],
+		}
+
+		err = stream.Send(response)
+		if err != nil {
+			logger.Error("unable to send fetch result", zap.Error(err))
```

Comment:
You will want to `return err` here if send fails, it's quite likely the connection is broken and should just terminate the send.